library/src/main/scala/scala/scalajs/js/Any.scala

```diff
@@ -28,22 +28,23 @@ import annotation.ScalaJSDefined
  *  [[AnyVal]]). Operations on JavaScript types behave as the corresponding
  *  operations in the JavaScript language.
  *
- *  By default, JavaScript types are native: they are facade types to APIs
- *  implemented in JavaScript code. Their implementation is irrelevant and
- *  never emitted. As such, all members must be defined with their
- *  right-hand-side being [[native js.native]]. For forward source
- *  compatibility with the next major version, the class/trait/object itself
- *  should be annotated with [[native @js.native]].
+ *  You can implement JavaScript types in Scala.js. The implementation (i.e. the
+ *  method and constructor bodies) will follow Scala semantics, but the
+ *  constructor and methods will be called using JavaScript semantics
+ *  (e.g. runtime dispatch).
+
+ *  A JavaScript type that is annotated with [[native @js.native]] is a facade
+ *  types to APIs implemented in JavaScript code. Their implementation is
+ *  irrelevant and never emitted. As such, all members must be defined with
+ *  their right-hand-side being [[native js.native]].
+ *  Further, native JavaScript types must be annotated with the name of the type
+ *  in JavaScript (using one of [[annotation.JSGlobal @JSGlobal]],
+ *  [[annotation.JSImport @JSImport]],
+ *  [[annotation.JSGlobalScope @JSGlobalScope]])
```

Comment:
"with the name of the type" is weird given that a) the name is optional in `@JSGlobal` and b) `@JSGlobalScope` doesn't even have a name. Maybe

> must be annotated with one of `@JSGlobal`, `@JSImport` and `@JSGlobalScope` to specify where to fetch it from.